tools/cgpatch/src/SymbolRetriever/SymbolRetriever.cpp

```diff
@@ -0,0 +1,299 @@
+/**
+ * File: SymbolRetriever.cpp
+ * License: Part of the MetaCG project. Licensed under BSD 3 clause license. See LICENSE.txt file at
+ * https://github.com/tudasc/metacg/LICENSE.txt
+ */
+
+#include "SymbolRetriever.h"
+
+#include <cstring>
+#include <elf.h>
+#include <fcntl.h>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <unistd.h>
+
+namespace SymbolRetriever {
+
+auto console = metacg::MCGLogger::instance().getConsole();
+auto errConsole = metacg::MCGLogger::instance().getErrConsole();
+
+struct RemoveEnvInScope {
+  explicit RemoveEnvInScope(const char* varName) : varName(varName) {
+    oldVal = getenv(varName);
+    if (oldVal) {
+      setenv(varName, "", true);
+    }
+  }
+
+  ~RemoveEnvInScope() {
+    if (oldVal) {
+      setenv(varName, oldVal, true);
+    }
+  }
+
+ private:
+  const char* varName;
+  const char* oldVal;
+};
+
+uintptr_t get_text_section_offset_from_library(const std::string& lib_path) {
```

Comment:
Specific reason here to break naming convention? Otherwise, please use camelCase